core-java-modules/core-java-date-operations-4/src/test/java/com/baeldung/longtodate/LongToDateUnitTest.java

```diff
@@ -0,0 +1,69 @@
+package com.baeldung.longtodate;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.TimeZone;
+
+import org.junit.jupiter.api.Test;
+
+class LongToDateUnitTest {
+
+    @Test
+    void givenLongValue_whenUsingDateClass_thenConvert() throws ParseException {
+        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+        dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
+        Date expectedDate = dateFormat.parse("2023-10-15 22:00:00");
+        long milliseconds = 1697407200000L;
+        Date date = new Date(milliseconds);
+
+        assertEquals(expectedDate, date);
+    }
+
+    @Test
+    void givenLongValue_whenUsingCalendarClass_thenConvert() throws ParseException {
+        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+        dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
+        Date expectedDate = dateFormat.parse("2023-07-15 22:00:00");
+
+        long milliseconds = 1689458400000L;
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTimeZone(TimeZone.getTimeZone("UTC"));
+        calendar.setTimeInMillis(milliseconds);
+
+        assertEquals(expectedDate, calendar.getTime());
+    }
+
+    @Test
+    void givenLongValue_whenUsingInstantClass_thenConvert() {
+        Instant expectedDate = Instant.parse("2020-09-08T12:16:40Z");
+        long seconds = 1599567400L;
+        Instant date = Instant.ofEpochSecond(seconds);
+
+        assertEquals(expectedDate, date);
+    }
+
+    @Test
+    void givenLongValue_whenUsingLocalDateClass_thenConvert() {
+        LocalDate expectedDate = LocalDate.of(2023, 10, 17);
+        long epochDay = 19647L;
+        LocalDate date = LocalDate.ofEpochDay(epochDay);
+
+        assertEquals(expectedDate, date);
+    }
+
+    @Test
+    void givenLongValue_whenUsingJodaTimeLocalDateClass_thenConvert() {
```

Comment:
This test could fail depending on the local time zone.

Reply:
Time zone added ;)